apps/documentation/src/routes/documentation/rendering/server-side-rendering.mdx

```diff
@@ -9,6 +9,50 @@ import Breadcrumbs from '@/components/Breadcrumbs'
 
 <Breadcrumbs breadcrumbs={[{ label: 'Rendering' }]} />
 
-# SSR
+# Server side rendering
 
-TODO
+## Overview
+
+If a page uses **Server-side Rendering**, the page HTML is generated on **each request**.
+
+> Every page is automatically server side rendered by default.
+
+To enable SSR on a page you just need to create the same file with the rust extension:
+
+- `src/routes/about.tsx`
+- `src/routes/about.rs`
+
+and then create an `#[tuono_lib::handler]` in the rust file that will be the
+function actually called on each request before rendering the HTML.
+
+## Example
+
+For example, suppose that your page needs to pre-render frequently updated data
+(read from a local file). You can write an hanlder which reads a json file and passes
+it to the route like below:
+
+```rs
+// src/routes/about.rs
+use tuono_lib::{Request, Response, Props};
+use std::fs;
+
+#[tuono_lib::handler]
+async fn my_custom_ssr_page(req: Request) -> Response {
+  let data = fs::read_to_string("my_data.json").expect("Failed to read json file");
+
+  Response::Props(Props::new(data))
```

Comment:
I haven't used this feature so far but I have some doubts about `Props` name usage:
I don't know how much familiar this term can be for user when dealing with http responses and with framework different from React.
Since you plan to make `tuono` framework agnostic I would consider another name.
I would use something tied to HTTP response statuses like
`OK` (200) since another option for Response is `Redirect`.
If this isn't ok here are a few additional alternatives:
- `Content`
- `Output`
- `RouteData`

(We can address this in a different PR)

Reply:
Agree. I'd propose to do an investigation about the overall DX and syntax! Agree that it is pretty rough right now